modules.d/90ppcmac/module-setup.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+#
+# This module attempts to properly deal with thermal behavior on PowerPC
+# based Mac systems, by installing the model-appropriate (when hostonly)
+# or all (when not) fan control/thermal kernel modules and loading them
+# in a hook.
+#
+# While this is not strictly necessary for all kernels, particularly
+# modular kernels will not autoload those drivers, even once the full
+# system is up, which results in the fans spinning up to 100%; this is
+# particularly annoying on live systems, where the system takes a while
+# to load, so it's best to load the drivers early in initramfs stage.
+#
+# The behavior of this is inspired by the thermal hook in Debian's
+# initramfs-tools, but written for dracut specifically and updated
+# for modern kernels (2012+).
+
+# called by dracut
+check() {
+    local _arch="$(uname -m)"
+    # only for PowerPC Macs
+    [[ "$_arch" == ppc* && "$_arch" != ppc*le ]] || return 1
+    return 0
```

Comment:
Shouldn't this also check for the pm_model and only return 0 if it matches?

Reply:
Well, no, because the idea is that non-hostonly initramfs images will have it regardless of the machine they're created on (which may be any `ppc64` or `ppc` system). For example, I use this in bootable live images, which are meant to be run on a variety of hardware. And even on a regular system, unless you set your dracut to hostonly, moving it to another hardware should result in everything still working. Which may be between different PowerMacs or even between a non-Apple and Apple system.

Reply:
ok